src/main/webapp/app/exam/participate/exam-participation.component.ts

```diff
@@ -888,4 +889,8 @@ export class ExamParticipationComponent implements OnInit, OnDestroy, ComponentC
                 }
             });
     }
+
+    asQuizExercise(exercise: Exercise): QuizExercise {
```

Comment:
Would it make sense to check if this is a valid cast?

Reply:
The check is expected to be done by the caller, which in this case is the *ngSwitchCase="QUIZ". This ensures that the exercise parameter is always of type QuizExercise.

Reply:
Sorry, I should've been more specific. That's precisely what I wanted to suggest.

Reply:
Type guard is added before casting